main.rb

```diff
@@ -0,0 +1,570 @@
+puts 'my_map: Test 2'
+print '[4, 6, 7, 9].map { |x| x / 2 > 3  }    #==>: '
+p [4, 6, 7, 9].map { |x| x / 2 > 3 }
+
+print '[4, 6, 7, 9].my_map { |x| x / 2 > 3 } #==>: '
+p [4, 6, 7, 9].my_map { |x| x / 2 > 3 }
+puts ''
+
+puts 'my_map: Test 3'
+print '[4, 6, 7, 9].map    #==>: '
+p [4, 6, 7, 9].map
+
+print '[4, 6, 7, 9].my_map #==>: '
+p [4, 6, 7, 9].my_map
+puts ''
+
+puts '==============  Point 9 #my_inject =============='
+puts 'my_any_full: Test 1'
+print 'p (5..10).inject { |sum, n| sum + n }    #==>: '
+p (5..10).inject { |sum, n| sum + n }
+
+print 'p (5..10).my_inject { |sum, n| sum + n } #==>: '
+p (5..10).my_inject { |sum, n| sum + n }
+puts ''
+
+puts 'my_any_full: Test 2'
+print 'p (5..10).inject(1) { |product, n| product * n }     #==>: '
+p (5..10).inject(1) { |product, n| product * n }
+
+print 'p (5..10).my_inject(1) { |product, n| product * n }  #==>: '
+p (5..10).my_inject(1) { |product, n| product * n }
+puts ''
+
+puts 'my_any_full: Test 3'
+print 'p %w{ cat sheep bear }.inject { |memo, word| memo.length > word.length ? memo : word }     #==>: '
+p %w[cat sheep bear].inject { |memo, word| memo.length > word.length ? memo : word }
+print 'p %w{ cat sheep bear }.my_inject { |memo, word| memo.length > word.length ? memo : word }  #==>: '
+p %w[cat sheep bear].my_inject { |memo, word| memo.length > word.length ? memo : word }
+puts ''
+
+puts '==============  POINT 10 - #my_inject test - multiply_else =============='
+puts 'mutilply_els using my_inject: Test 1'
+print 'p multiply_els([2, 4, 5])  #==>: '
+p multiply_els([2, 4, 5])
+puts ''
+
+puts '==============  POINT 11/12 - #my_map modified =============='
+
+my_proc = proc { |arg1| arg1 / 2 > 3 }
+
+puts 'my_map: Test using normal block'
+print '[4, 6, 7.0, 9].my_map { |x| x / 2 > 3  }    #==>: '
+p [4, 6, 7.0, 9].my_map { |x| x / 2.0 > 3.0 }
+puts ''
+
+puts 'my_map: Test using proc & block - Executing Proc as priority'
+print '[4, 6, 7.0, 9].my_map_proc(my_proc) {|arg1| arg1 / 2.0 > 3.0 } #==>: '
+p [4, 6, 7.0, 9].my_map(my_proc) { |arg1| arg1 / 2.0 > 3.0 }
+
+puts '==============  My Inject_proc =============='
+
+print '[10, 20, 30, 40, 50].inject(:+) #==>: '
+p [10, 20, 30, 40, 50].inject(:+)
+print '[10, 20, 30, 40, 50].my_inject(:+) #==>: '
+p [10, 20, 30, 40, 50].my_inject(:+)
+puts ''
+
+print '[1, 2, 3, 4, 5].inject(:*) #==>: '
+p [1, 2, 3, 4, 5].inject(:*)
+print '[1, 2, 3, 4, 5].my_inject(:*) #==>: '
+p [1, 2, 3, 4, 5].my_inject(:*)
+puts ''
+
+print '[1, 2, 3, 4, 5].inject(:/) #==>: '
+p [1, 2, 3, 4, 5].inject(:/)
+print '[1, 2, 3, 4, 5].my_inject(:/) #==>: '
+p [1, 2, 3, 4, 5].my_inject(:/)
+puts ''
+
+print '[10, 20, 30, 40, 50].inject(5,:+) #==>: '
+p [10, 20, 30, 40, 50].inject(5, :+)
+print '[10, 20, 30, 40, 50].my_inject(5,:+) #==>: '
+p [10, 20, 30, 40, 50].my_inject(5, :+)
+puts ''
+
+print '[1, 2, 3, 4, 5].inject(5,:*) #==>: '
+p [1, 2, 3, 4, 5].inject(5, :*)
+print '[1, 2, 3, 4, 5].my_inject(5,:*) #==>: '
+p [1, 2, 3, 4, 5].my_inject(5, :*)
+puts ''
```

Comment:
- Please remove this code, you could place it on a separate file for testing purposes

Reply:
I now have moved the tests to a new file.